CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+  var authorType = [ "creator", "contributor" ]; // Author field names in the mapping table must be Zotero author types
+  for ( var i in authorType ) {
+	if ( item[ authorType[ i ] ] ) {
+	  fetchedAuthor = item[ authorType[ i ] ].replace( /\([^)]*\)|\[[^\]]*\]|\;$/g, '' ).replace( /, Sir |, Dr /g, ', ' ); // Fetch and remove parentheses and honorifics
+	  var nameArray = [];
+	  item[ authorType[ i ] ] = "";
+	  nameArray = fetchedAuthor.split( ";" ); // Split authors at semicolon and push into an array
+	  for ( var j in nameArray ) item.creators.push( ZU.cleanAuthor( nameArray[ j ], authorType[ i ], 1 ) );
+	}
+  }
+  for ( var k in item.creators ) {
+	if ( !item.creators[ k ].firstName ) item.creators[ k ].fieldMode = 1;
+  }
+}
+
+function attachPDF( doc, item ) {
+  var pdfPath = '//embed';
+  var pdfScrape = ZU.xpath( doc, pdfPath );
+  var pdfObject = pdfScrape[ 0 ]; // Could iterate looking for multiple attachments
+  var pdfTitle;
+  if ( pdfObject && pdfObject.src ) { // Can't simply test for .src
+	var itemTitle = fetchXPathText( doc, '//*[@id="metadata_title"]' );
+	if ( itemTitle ) pdfTitle = "Attachment: " + itemTitle;
+	else pdfTitle = "Attachment";
+	item.attachments.push( {
+	  url: pdfObject.src,
+	  title: pdfTitle,
+	  mimeType: "application/pdf"
+	} );
+  }
+}
+
+function doWeb( doc, url ) {
+  if ( detectWeb( doc, url ) == "multiple" ) { // If on a multiple result page, call Zotero.selectItems
+	var titlePath = '//div/ul/li/ul/li[2]/div/a';
+	var items = {};
+	var articles = [];
+	var titles = ZU.xpath( doc, titlePath );
+	if ( titles.length < 1 ) return false; // Null search results will be ignored
+	for ( var i in titles ) {
+	  items[ titles[ i ].href ] = ZU.trimInternal( titles[ i ].textContent );
+	}
+	Zotero.selectItems( items, function( items ) {
+	  if ( !items ) return true;
+	  for ( var i in items ) {
+		articles.push( i );
+	  }
+	  ZU.processDocuments( articles, scrape );
+	} );
+
+  } else if ( url.match( "/cdm/fullbrowser" ) ) {
+	var originatingItem = ZU.xpath( doc, '//link[@rel="canonical"]' ); // Look for link to originating item
+	if ( originatingItem[ 0 ] && originatingItem[ 0 ].href ) {
+	  url = originatingItem[ 0 ].href;
+	  ZU.processDocuments( url, scrape );
+	} else return false;
+
+  } else if ( ZU.xpath( doc, '//input[@id="cdm_newRefUrl"]' ) ) {
+	var realItem = ZU.xpath( doc, '//input[@id="cdm_newRefUrl"]' ); // CDM pages can display info for other items; look for real url of item
+	if ( realItem[ 0 ] && realItem[ 0 ].value ) {
+	  url = realItem[ 0 ].value;
+	  ZU.processDocuments( url, scrape );
+	} else scrape( doc, url );
+
+  } else {
+	scrape( doc, url );
+  }
+}
+
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "http://cdm15979.contentdm.oclc.org/cdm/compoundobject/collection/p15979coll3/id/2419",
+		"items": [
+			{
+				"itemType": "manuscript",
+				"title": "MS.15.1.2.000. Sir Robert Hart Diary: Volume 02: February 1855-July 1855",
```

Comment:
There are two points here. 1) In this particular catalogue record, 'MS.15.1.2.000' is given as the item title, hence it is scraped by the translator. On this occasion, the item title is redundant, but in many cases it isn't. 

2) The slightly awkward format is because CDM 6 has both object-level data and item-level data. These are implemented inconsistently by libraries, and it is not possible to say that either can be safely omitted. Because of the way CDM displays, libraries often choose to put essential information at the Object level, and (as in this example) simply dropping the object-level data if the item data is present would lead to some uninformative scrapes. Hence the translator combines the item information and the object level data.